crates/rbuilder/src/live_builder/base_config.rs

```diff
@@ -602,29 +602,46 @@ pub fn coinbase_signer_from_secret_key(secret_key: &str) -> eyre::Result<Signer>
     Ok(Signer::try_from_secret(secret_key)?)
 }
 
-fn get_signing_domain(chain: Chain, cl_network_config_dir: Option<String>) -> eyre::Result<B256> {
-    let network = named_chain_to_network(chain)
-        .or_else(|| cl_network_config_dir.map(Network::Custom))
-        .ok_or_else(|| {
-            eyre::eyre!("Unknown chain. If network is not named, specify cl_network_config_dir")
-        })?;
+fn get_signing_domain(chain: Chain, beacon_clients: Vec<Client>) -> eyre::Result<B256> {
+    let cl_context = match chain.kind() {
+        ChainKind::Named(NamedChain::Mainnet) => ContextEth::for_mainnet(),
+        ChainKind::Named(NamedChain::Sepolia) => ContextEth::for_sepolia(),
+        ChainKind::Named(NamedChain::Goerli) => ContextEth::for_goerli(),
+        ChainKind::Named(NamedChain::Holesky) => ContextEth::for_holesky(),
+        _ => {
+            let rt = Runtime::new()?;
```

Comment:
It's illegal to create nested runtimes, were you able to run this from a tokio main?

Reply:
I was able to run this from a tokio::test.

Reply:
The final solution is to admit that this is a blocking call and propagate the async, it's not nice to block too much on tokio unless you are on a tokio::spawn_blocking or Godzilla will come out and kick your ass.